src/import/engine.rs

```diff
@@ -121,16 +121,18 @@ impl Engine {
         let collection = self.get_properties_of_all_tables()?;
         for (_, v) in &*collection {
             let props = RangeProperties::decode(v.user_collected_properties())?;
-            res.total_size += props.offsets.iter().map(|pair| pair.1.size).sum::<u64>();
-            res.index_handles.extend(props.offsets.iter().map(|pair| {
-                (
-                    pair.0.clone(),
+            let mut prev_size = 0;
+            for (key, range) in props.offsets {
+                res.index_handles.add(
+                    key,
                     IndexHandle {
-                        size: pair.1.size,
-                        offset: 0,
+                        size: range.size - prev_size,
+                        offset: range.size,
                     },
-                )
-            }));
+                );
+                prev_size = range.size;
+            }
+            res.total_size += prev_size;
```

Comment:
If `range.size` is a cumulative size, I guess `res.total_size` should equal to `range.size`.